Desktop_Interface/uartstyledecoder.cpp

```diff
@@ -2,41 +2,43 @@
 #include <QDebug>
 #include <cassert>
 
-uartStyleDecoder::uartStyleDecoder(QObject *parent_in) : QObject(parent_in)
+uartStyleDecoder::uartStyleDecoder(QObject *parent_in)
+	: QObject(parent_in)
```

Comment:
nit: `m_parent` and `parent` would be better names than `parent` and `parent_in`.

That said, I don't think we should store parent at all (see below).